### ISSUE TITLE
[Julia API] Query returns AbstractString instead of String

### ISSUE THREAD

**Issue by jfb-h** [duckdb/duckdb#4801]
Using the Julia API, queries including strings return as `AbstractString` but should probably be some concrete string type, such as `String`. Here's an example to reproduce (on 0.5.1):

```julia
using DuckDB, DBInterface
con = DBInterface.connect(DuckDB.DB, ":memory:")
DBInterface.execute(con, "CREATE TABLE test(a INT, b VARCHAR);")
DBInterface.execute(con, "INSERT INTO test VALUES (1, 'abc');")
df = DBInterface.execute(con, "SELECT * FROM test").df
typeof(df.b) # Vector{Union{Missing, AbstractString}}
```

**Comment by hannes** [comment]
Could you please use the issue template? Thanks.

**Comment by jfb-h** [comment]
Sorry, closed in favour of #4809 which uses the template.
